compat/maven-compat/src/main/java/org/apache/maven/artifact/repository/layout/FlatRepositoryLayout.java

```diff
@@ -62,19 +62,11 @@ public String pathOf(Artifact artifact) {
     }
 
     public String pathOfLocalRepositoryMetadata(ArtifactMetadata metadata, ArtifactRepository repository) {
-        return pathOfRepositoryMetadata(metadata.getLocalFilename(repository));
-    }
-
-    private String pathOfRepositoryMetadata(String filename) {
-        StringBuilder path = new StringBuilder(128);
-
-        path.append(filename);
-
-        return path.toString();
+        return metadata.getLocalFilename(repository);
     }
 
     public String pathOfRemoteRepositoryMetadata(ArtifactMetadata metadata) {
-        return pathOfRepositoryMetadata(metadata.getRemoteFilename());
+        return metadata.getRemoteFilename();
```

Comment:
why no default impl in interface its pure domain logic?
- https://github.com/apache/maven/pull/2402